src/uwtools/tests/test_config.py

```diff
@@ -810,3 +811,46 @@ def test_YAMLConfig__load_unexpected_error(tmp_path):
         with raises(UWConfigError) as e:
             config.YAMLConfig(config_path=cfgfile)
         assert msg in str(e.value)
+
+
+def test_print_config_section_yaml(capsys):
+    config_obj = config.YAMLConfig(fixture_path("FV3_GFS_v16.yaml"))
+    section = ["sgs_tke", "profile_type"]
+    config.print_config_section(config_obj.data, section, log=Logger())
+    actual = capsys.readouterr().out
+    expected = """name=fixed
+surface_value=0.0\n"""
+    assert actual == expected
+
+
+def test_print_config_section_yaml_for_nonscalar():
+    config_obj = config.YAMLConfig(fixture_path("FV3_GFS_v16.yaml"))
+    section = ["o3mr"]
+    with raises(UWConfigError):
+        config.print_config_section(config_obj.data, section, log=Logger())
+
+
+def test_print_config_section_yaml_not_dict():
+    config_obj = config.YAMLConfig(fixture_path("FV3_GFS_v16.yaml"))
+    section = ["sgs_tke", "units"]
+    with raises(UWConfigError):
+        config.print_config_section(config_obj.data, section, log=Logger())
+
+
+def test_print_config_section_ini(capsys):
+    config_obj = config.INIConfig(fixture_path("simple3.ini"))
+    section = ["dessert"]
+    config.print_config_section(config_obj.data, section, log=Logger())
+    actual = capsys.readouterr().out
+    expected = """type=pie
+flavor={{flavor}}
+side=False
+servings=0\n"""
+    assert actual == expected
+
+
+def test_print_config_section_ini_missing_section():
+    config_obj = config.INIConfig(fixture_path("simple3.ini"))
+    section = ["sandwich"]
+    with raises(KeyError):
+        config.print_config_section(config_obj.data, section, log=Logger())
```

Comment:
Nice tests!

A couple small ideas:

1. In the second code box in [Assertions about expected exceptions](https://docs.pytest.org/en/7.1.x/how-to/assert.html#assertions-about-expected-exceptions) the `pytest` docs show how to assign the caught exception to a variable and then make an assertion about its text content. Since you give useful messages in the `UWConfigError` exceptions you raise, you could potentially strengthen for `_for_nonscalar()` and `_not_dict()` tests to look for e.g. `"must be scalar"` and `"must be a dictionary"`, respectively. That would rule out any chance that the test was catching the _other_ exception.

2. I think you might be able to set `expected` values like e.g.
```
expected = """
name=fixed
surface_value=0.0
""".lstrip()
```
if that seems any better to you. That makes the `expected` value maybe a little easier to read and should do the right thing with the newlines. I think... I do not feel strongly about this.

3. Consider sorting the `_ini*()` functions above the `_yaml*()` ones?